### PR TITLE
implement ppc_fp128

### PR DESCRIPTION
**Pull request by dannypsnl**

### CODE REVIEW

File: float128ppc/float128ppc.go [hunk 1]
```diff
@@ -1,50 +1,106 @@
...
-	//    112 bits: fraction
-	//    8 bits:   unused
-	a uint64
-	b uint64
+	// where a long double value is regarded as the exact sum of two double-precision values, giving at least a 106-bit precision
+	a float64
+	b float64
 }
 
 // NewFromBits returns the floating-point number corresponding to the IBM
```
Comment (mewmew) [inline float128ppc/float128ppc.go]:
```diff
-// NewFromBits returns the floating-point number corresponding to the IBM
-// extended double representation.
+// NewFromBits returns the floating-point number corresponding to the
+// double-double representation.
```

File: float128ppc/float128ppc.go [hunk 2]
```diff
@@ -1,50 +1,106 @@
...
 // extended double representation.
 func NewFromBits(a, b uint64) Float {
-	return Float{a: a, b: b}
+	return Float{a: math.Float64frombits(a), b: math.Float64frombits(b)}
 }
 
 // NewFromFloat32 returns the nearest IBM extended double floating-point number
```
Comment (mewmew) [inline float128ppc/float128ppc.go]:
```diff
-// NewFromFloat32 returns the nearest IBM extended double floating-point number
-// for x and a bool indicating whether f represents x exactly.
+// NewFromFloat32 returns the nearest double-double precision floating-point
+// number for x and the accuracy of the conversion.
```

File: float128ppc/float128ppc.go [hunk 3]
```diff
@@ -1,50 +1,106 @@
...
-	panic("not yet implemented")
+func NewFromFloat32(x float32) (f Float, exact big.Accuracy) {
+	f, acc := NewFromFloat64(float64(x))
+	if acc == big.Exact {
+		_, acc = f.Float32()
+	}
+	return f, acc
 }
 
 // NewFromFloat64 returns the nearest IBM extended double floating-point number
```
Comment (mewmew) [inline float128ppc/float128ppc.go]:
```diff
-// NewFromFloat64 returns the nearest IBM extended double floating-point number
-// for x and a bool indicating whether f represents x exactly.
+// NewFromFloat64 returns the nearest double-double precision floating-point
+// number for x and the accuracy of the conversion.
```

File: float128ppc/float128ppc.go [hunk 4]
```diff
@@ -1,50 +1,106 @@
...
+		}
+		// +NaN
+		return NaN, big.Exact
+	}
+	r := Float{a: x, b: 0}
+	br, _ := r.Big()
+	return r, br.Acc()
 }
 
 // Bits returns the IBM extended double binary representation of f.
```
Comment (mewmew) [inline float128ppc/float128ppc.go]:
```suggestion
// Bits returns the double-double binary representation of f.
```

File: float128ppc/float128ppc.go [hunk 5]
```diff
@@ -1,50 +1,106 @@
...
+// a boolean indicating whether f is Not-a-Number.
+func (f Float) Big() (x *big.Float, nan bool) {
+	x = big.NewFloat(0)
+	x.SetPrec(precision)
+	x.SetMode(big.ToNearestEven)
+	if math.IsNaN(f.a) || math.IsNaN(f.b) {
+		return x, true
+	}
+	h := big.NewFloat(f.a)
+	l := big.NewFloat(f.b)
```
Comment (mewmew) [inline float128ppc/float128ppc.go]: We may want to consider renaming `f.a` to `f.h` (or `f.hi`) and `f.b` to `f.l` (or `f.lo`), as that better captures the names and semantics of double-double representation.

**Review by mewmew — COMMENTED** [review]
LGTM with minor comments.

File: float128ppc/float128ppc.go [hunk 6]
```diff
@@ -16,25 +16,25 @@ const (
...
 
 var (
-	NegNaN = Float{a: -math.NaN(), b: 0}
-	NaN    = Float{a: math.NaN(), b: 0}
+	NegNaN = Float{high: -math.NaN(), low: 0}
+	NaN    = Float{high: math.NaN(), low: 0}
 )
 
-// Float is a floating-point number in double-double format.
+// Float is high floating-point number in double-double format.
```
Comment (mewmew) [inline float128ppc/float128ppc.go]: it seems like `a` was replaced with `high` in too many places ^^

File: float128ppc/float128ppc.go [hunk 7]
```diff
@@ -16,25 +16,25 @@ const (
...
 type Float struct {
-	// where a long double value is regarded as the exact sum of two double-precision values, giving at least a 106-bit precision
-	a float64
-	b float64
+	// where high long double value is regarded as the exact sum of two double-precision values, giving at least high 106-bit precision
```
Comment (mewmew) [inline float128ppc/float128ppc.go]: high -> a

File: float128ppc/float128ppc.go (hunk 6 above)
Comment (dannypsnl) [reply float128ppc/float128ppc.go]: Oh damn, this feature of GoLand sometimes annoying when naming variable too casual XD.

**Review by mewmew — APPROVED** [review]
LGTM.

**Comment by mewmew** [comment]
Thanks for researching and implementing this @dannypsnl!

**Pull request merged**